src/epanet.c

```diff
@@ -819,6 +819,13 @@ int DLLEXPORT EN_closeQ(EN_Project p)
     if (!p->Openflag) return 102;
     closequal(p);
     p->quality.OpenQflag = FALSE;
+    
+    if (p->outfile.OutFile != NULL)
+    {
+        fclose(p->outfile.OutFile);
```

Comment:
I would suggest checking if TmpOutFile is the same as OutFile where
```suggestion
        if (p->outfile.OutFile == p->outfile.TmpOutFile){
            p->outfile.TmpOutFile = NULL;
        }
        fclose(p->outfile.OutFile);
```

Reply:
Your fix works! I re-factored closing the output file into its own function `closeoutfile` in project .c since the same code is used both in `EN_closeQ` and in `openoutfile`. Thank you @gonccalo !